social_engagement/tasks.py

```diff
@@ -5,66 +5,46 @@
 import pytz
 from datetime import datetime
 
+from django.conf import settings
 from django.contrib.auth.models import User
 from django.db.models import F
 from celery.task import task
 
-from student.models import CourseEnrollment
 from xmodule.modulestore.django import modulestore
 from opaque_keys.edx.keys import CourseKey
 
-from social_engagement.engagement import update_user_engagement_score, get_social_metric_points
+from social_engagement.engagement import update_course_engagement, get_social_metric_points
 from social_engagement.models import StudentSocialEngagementScore
 
 log = logging.getLogger('edx.celery.task')
 
 
-@task(name=u'lms.djangoapps.social_engagement.tasks.task_update_user_engagement_score')
-def task_update_user_engagement_score(course_id, user_id):
-    """
-    Task to update user's engagement score
-    """
-    try:
-        update_user_engagement_score(course_id, user_id)
-    except Exception as exc:   # pylint: disable=broad-except
-        log.info(
-            "social engagement score update failure for course {} and user id {} with exception {}".format(
-                course_id,
-                user_id,
-                repr(exc)
-            )
-        )
-
-
-@task(name=u'lms.djangoapps.social_engagement.tasks.task_compute_social_scores_in_course')
+@task(
+    name=u'lms.djangoapps.social_engagement.tasks.task_compute_social_scores_in_course',
+    routing_key=settings.RECALCULATE_SOCIAL_ENGAGEMENT_ROUTING_KEY,
+)
 def task_compute_social_scores_in_course(course_id):
     """
     Task to compute social scores in course
     """
     score_update_count = 0
     course_key = CourseKey.from_string(course_id)
     course = modulestore().get_course(course_key, depth=None)
-    user_ids = CourseEnrollment.objects.values_list('user_id', flat=True).filter(
-        is_active=1,
-        course_id=course_key
-    )
 
     if course:
         # For each user compute and save social engagement score
-        for user_id in user_ids:
-            update_user_engagement_score(
-                course_key, user_id, compute_if_closed_course=True, course_descriptor=course
-            )
-            score_update_count += 1
+        score_update_count = update_course_engagement(
+            course_key, compute_if_closed_course=True, course_descriptor=course
+        )
     else:
         log.info("Course with course id %s does not exist", course_id)
     log.info("Social scores updated for %d users in course %s", score_update_count, course_id)
```

Comment:
I get an error here now. Apparently `score_update_count` was None.

Reply:
Am looking into why by stepping into the debugger

Reply:
Ah, right. Fixed!

Reply:
Ah, just saw your most recent change. Going to try again then.

Reply:
It might have been one of these three things:
- you don't have `ENABLE_SOCIAL_ENGAGEMENT` set,
- `course_descriptor` was `None` (I'm not sure why it might happen, I was there before),
- Forum API was not launched.

`update_course_engagement` would return `None` then and string parsing required `%d` there. It will not be an issue now.

Reply:
Btw. I've set this in `private.py`, so I don't have to change `ENABLE_SOCIAL_ENGAGEMENT` anymore (which I was always forgetting about):
```python
from django.conf import settings
settings.FEATURES['ENABLE_SOCIAL_ENGAGEMENT'] = True
```

Reply:
Shoot, yeah, I didn't have `ENABLE_SOCIAL_ENGAGEMENT` set... I need to set this in private.py too, right now I'm git-stashing/unstashing everytime I need to pull new changes.

Reply:
That actually turned out to be good this time, because we could track this bug :slightly_smiling_face: